### PR TITLE
tests: adc_api: Use new DT_LABEL macro for getting device name

### PR DESCRIPTION
**Pull request by galak**
Replace dts_fixup.h based DT_ADC_0_NAME defines with DT_LABEL as we
phase out dts_fixup.h

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

**Review by nandojve — APPROVED** [review]
Atmel SAM

**Review by dcpleung — APPROVED** [review]

**Review by anangl — APPROVED** [review]

File: tests/drivers/adc/adc_api/src/test_adc.c [hunk 1]
```diff
@@ -29,7 +29,7 @@
 #elif defined(CONFIG_BOARD_NRF51DK_NRF51422)
 
 #include <hal/nrf_adc.h>
-#define ADC_DEVICE_NAME		DT_ADC_0_NAME
+#define ADC_DEVICE_NAME		DT_LABEL(DT_INST(0, nordic_nrf_adc))
```
Comment (henrikbrixandersen) [inline tests/drivers/adc/adc_api/src/test_adc.c:32]: How can we be sure that `DT_INST(0, nordic_nrf_adc)` is `ADC_0`? Shouldn't we use `DT_NODELABEL()` here?

Comment (galak) [reply tests/drivers/adc/adc_api/src/test_adc.c:32]: I'm guessing in this case its know that ADC is a singleton, since the driver uses DT_INST.

Comment (henrikbrixandersen) [reply tests/drivers/adc/adc_api/src/test_adc.c:32]: It was more of a general comment to the conversions performed in this PR. Some of the coversions convert from ADC_1 to DT_INST(0, ...).

Comment (mbolivar-nordic) [reply tests/drivers/adc/adc_api/src/test_adc.c:32]: If the driver is multi-instance at all, the INST ordering should be assumed to be basically random, so this is indeed an issue in general (though for this particular line of source it's a singleton).

Comment (carlescufi) [reply tests/drivers/adc/adc_api/src/test_adc.c:32]: @mbolivar-nordic I might have been a bit too trigger-happy with this one. @henrikbrixandersen please feel free to continue the conversation here and then we can create a separate PR if necessary to tweak whatever is needed.

Comment (mbolivar-nordic) [reply tests/drivers/adc/adc_api/src/test_adc.c:32]: I personally think the INST conversions should be removed and replaced with either NODELABEL or ALIAS.

Comment (henrikbrixandersen) [reply tests/drivers/adc/adc_api/src/test_adc.c:32]: I agree.

Comment (galak) [reply tests/drivers/adc/adc_api/src/test_adc.c:32]: I'd prefer NODELABEL

Comment (galak) [reply tests/drivers/adc/adc_api/src/test_adc.c:32]: or in this case, for a lot of these PATH would make sense since they are board specific ifdefs.

Comment (mbolivar-nordic) [reply tests/drivers/adc/adc_api/src/test_adc.c:32]: Either NODELABEL or PATH seems fine as way to resolve this issue

**Pull request merged**